src/error.ts

```diff
@@ -0,0 +1,54 @@
+import { Response as ExResponse, Request as ExRequest, NextFunction } from 'express'
+import { ValidateError } from 'tsoa'
+
+import { Logger } from '@aries-framework/core'
+
+export class HttpResponse extends Error {
+  public code: number
+  public message: string
+
+  constructor({ code = 500, message = 'Unexpected error' }) {
+    super(message)
+    this.code = code
+    this.message = message
+    this.name = 'Internal server error'
+  }
+}
+
+export class NotFound extends HttpResponse {
+  constructor(message = 'not found') {
+    super({ code: 404, message })
+  }
+}
+
+export class BadRequest extends HttpResponse {
+  constructor(message = 'bad request') {
+    super({ code: 400, message })
+  }
+}
+
+export const errorHandler =
+  (logger: Logger) =>
+  (err: unknown, req: ExRequest, res: ExResponse, next: NextFunction): ExResponse | void => {
+    if (err instanceof ValidateError) {
+      logger.warn(`Caught Validation Error for ${req.path}:`, err.fields)
+      return res.status(422).json({
+        message: 'Validation Failed',
+        details: err?.fields,
+      })
+    }
+
+    if (err instanceof HttpResponse) {
+      logger.warn(`Error thrown in handler: ${err.message}`)
+
+      return res.status(err.code).json(err.message)
+    }
+    if (err instanceof Error) {
+      logger.error(`Unexpected error thrown in handler: ${err.message}`)
```

Comment:
in the original code it was like this:  
```
 agent.config.logger.error('Internal Server Error.', err)
      return res.status(500).json({
        message: 'Internal Server Error. Check server logging.',
      })
    }
```

I was just wondering do we no longer include 'Internal server error' we just say 'unexpected error in handler '? Are these two basically the same/interchangeable?

Reply:
They seem interchangeable to me. The new method actually returns more info to the user, maybe too much cause it includes the stack
![image](https://github.com/digicatapult/veritable-cloudagent/assets/40722025/88b803d1-5044-47fd-974c-4f00ce301a09)